### ISSUE TITLE
Can you perform a binary search a doubly linked list?

### ISSUE THREAD

**Issue by cathyatseneca** [SENECA-DSA555/dsa555-f16#9]

**Comment by Leafa200** [comment]
No, because there is no way to access the middle node of the list, even if the list is sorted.

**Comment by barefoot1423** [comment]
int iterative_GCD(int x, int y) {
    while (y != 0) {
        int r = x % y;
        x = y;
        y = r;
    }
    return x;
}


// Recursive
int recursive_GCD(int x, int y) {
    return (y == 0) ? (x) : (recursive_GCD(y, x % y));
}
